server/src/userManagement.mjs

```diff
@@ -236,11 +236,22 @@ function addUser(userId) {
     ws.on('pong', async hb => {
       heartbeat(ws)
     });
-    associateUserWithWs(''+userId, ws);
-    handleGroupMembership(''+userId)
-    ws.on('message', async message => {
-      messageHandler.handleMessage(message, ''+userId, ws);
-    });
+    // If multiUserConnections configuration is set as deny and there is a ws object associated with userId, deny and log second ws connection and drop the attempt
+    if (/deny/i.test(config.multiUserConnections) == true && getWsForUser(userId) !== undefined) {
```

Comment:
"!== undefined" is fine if you're *sure* it will always be undefined when, uh, undefined and not, say, null. A slightly less strict check would simply be to test the falsiness of getWsForUser(userId)... E.g., just use "!getWsForUser(userId)" instead.

Reply:
getWsForUser method was implemented in such a way it always returns 'undefined' if there is no ws object associated with a userid, hence the check
```
function getWsForUser(userId) {
  if ( user2ws.has(''+userId) ) {
    let wsArray=user2ws.get(''+userId);
    let latestWsConnection=wsArray[(wsArray.length-1)]
    return latestWsConnection;
  }
  return undefined;
}
```